src/include/txn.i

```diff
@@ -1091,12 +1105,24 @@ __wt_txn_update_check(WT_SESSION_IMPL *session, WT_UPDATE *upd)
         if (upd->txnid != WT_TXN_ABORTED) {
             if (ignore_prepare_set)
                 F_SET(txn, WT_TXN_IGNORE_PREPARE);
-            WT_STAT_CONN_INCR(session, txn_update_conflict);
-            WT_STAT_DATA_INCR(session, txn_update_conflict);
-            return (__wt_txn_rollback_required(session, "conflict between concurrent operations"));
+            return (__txn_rollback_required(session));
         }
     }
 
+    /* If there is no cursor, we don't need to check the on page value. */
+    WT_ASSERT(session, cbt != NULL || (cbt == NULL && !check_onpage_value));
```

Comment:
Can't this be written as `cbt != NULL || !check_onpage_value`?

Reply:
fixed